services/httpd/handler.go

```diff
@@ -662,9 +661,9 @@ func gzipFilter(inner http.Handler) http.Handler {
 
 // versionHeader taks a HTTP handler and returns a HTTP handler
 // and adds the X-INFLUXBD-VERSION header to outgoing responses.
-func versionHeader(inner http.Handler, version string) http.Handler {
+func versionHeader(inner http.Handler, h *Handler) http.Handler {
 	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
-		w.Header().Add("X-InfluxDB-Version", version)
+		w.Header().Add("X-InfluxDB-Version", h.Version)
```

Comment:
actually the reason for it is that at the time `versionHeader` is called `h.Version` has not been set. so I passed a pointer to `h` and by the time the `HandlerFunc` is called, `h.Version` has been set. there were alternatives to this but I found this one to be my preference.